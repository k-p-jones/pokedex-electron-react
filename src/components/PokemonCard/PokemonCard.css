.pokemon-card {
  margin-bottom: 20px;
  padding: 10px;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
}

.pokemon-card-text {
  margin-top: 5px;
  text-transform: uppercase;
}
