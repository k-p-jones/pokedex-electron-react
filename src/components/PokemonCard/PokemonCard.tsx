import React from 'react';
import { Grid, Image } from 'semantic-ui-react';
import COLORS from '../../consts/colors';
import Pokemon from '../../interfaces/Pokemon';
import './PokemonCard.css';

interface Props {
  pokemon: Pokemon;
}

const PokemonCard: React.FC<Props> = ({ pokemon }) => {
  const textColor: string = pokemon.color === COLORS['white'] ? COLORS['altText'] : COLORS['white'];

  return (
    <Grid.Column>
      <div className="pokemon-card" style={{ backgroundColor: pokemon.color, color: textColor}}>
        <div>
          <Image
            src={pokemon.imageURL}
          />
        </div>
        <h3 style={{ marginTop: 5, textTransform: 'uppercase' }} id={`header-${pokemon.name}`} >{pokemon.name}</h3>
      </div>
    </Grid.Column>
  )
}

export default PokemonCard;
