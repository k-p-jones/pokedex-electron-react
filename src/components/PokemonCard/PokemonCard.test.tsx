import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PokemonCard from './PokemonCard';

const pokemon = {
  id: 1,
  name: 'bulbasaur',
  speciesURL: 'https://pokeapi.co/api/v2/pokemon-species/1',
  dataURL: 'https://pokeapi.co/api/v2/pokemon/1',
  imageURL: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
  data: {
    types: [],
    stats: []
  },
  color: 'green',
  evolutionChainURL: 'https://pokeapi.co/api/v2/evolution-chain/1',
  dataFetched: false,
  typeDataFetched: false,
  evolutionChainFetched: false
}

test('it renders correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <PokemonCard pokemon={pokemon} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
})
