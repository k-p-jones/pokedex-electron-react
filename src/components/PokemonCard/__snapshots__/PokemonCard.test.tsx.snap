// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    class="column"
  >
    <a
      href="/pokemon"
    >
      <div
        class="pokemon-card"
        style="background-color: green; color: rgb(255, 255, 255);"
      >
        <img
          class="ui image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
        />
        <h3
          class="pokemon-card-text"
          id="header-bulbasaur"
        >
          bulbasaur
        </h3>
      </div>
    </a>
  </div>
</div>
`;
